### ISSUE TITLE
backend equivalent of length() to use in a custom loss function

### ISSUE THREAD

**Issue by reyhanhgh** [rstudio/keras#1060]
Hi !
I would like to create a custom `loss` function. To do this, I need to get the number of elements in `y_true` ( that is equal to `y_pred`). What is the `backend` equivalent of ` length()` to do so ? for preparing my input data, I'am using the generator proposed by F. Chollet & J. J. Allaire in [this tutorial](https://blogs.rstudio.com/ai/posts/2017-12-20-time-series-forecasting-with-recurrent-neural-networks/). Your help would be appreciated.

**Comment by reyhanhgh** [comment]
It seems that ` k_shape(y_true)` returns the size of `y_true` but since it is an integer I am supposed to convert it to a float array. but when  i tried ` k_floatx(k_shape(y_true)[1])` it gives an error. The only form that it worked when I compiled the model was `k_cast_to_floatx(k_shape(y_true))`. But when I tried to trained the model it gave `nan` as loss for all epochs. Here is the entire custom loss function that I created :
loss <- function(y_true, y_pred) {
  
  alpha <- (k_std(y_pred))/(k_std(y_true))
  beta  <- (k_mean(y_pred))/(k_mean(y_true))

  r1 <- y_pred - k_mean(y_pred)
  r2 <- y_true - k_mean(y_true)
  sum_rr <- k_sum(r1*r2)
  n <- k_cast_to_floatx(k_shape(y_true))
  rcv <- sum_rr/(n-1)

  r <- rcv/(k_std(y_pred) * k_std(y_true))
  
  L <- k_sqrt( k_pow((1-alpha), 2) + k_pow((1-beta), 2) + k_pow((1-r), 2) )

  return(L)
}
